activemodel/lib/active_model/enum.rb

```diff
@@ -0,0 +1,347 @@
+  #     attribute :status, :integer
+  #     enum :status, [ :active, :archived ]
+  #   end
+  #
+  #   conversation.active!
+  #   conversation.active? # => true
+  #   conversation.status  # => "active"
+  #
+  #   conversation.archived!
+  #   conversation.archived? # => true
+  #   conversation.status    # => "archived"
+  #
+  #   conversation.status = nil
+  #   conversation.status.nil? # => true
+  #   conversation.status      # => nil
+  #
+  # You can set the default enum value by setting +:default+, like:
+  #
+  #   class Conversation
+  #     include ActiveModel::Attributes
+  #     include ActiveModel::Enum
+  #     attribute :status, :integer
+  #     enum :status, [ :active, :archived ], default: :active
+  #   end
+  #
+  #   conversation = Conversation.new
+  #   conversation.status # => "active"
+  #
+  # It's possible to explicitly map the relation between attribute and
+  # database integer with a hash:
+  #
+  #   class Conversation
+  #     include ActiveModel::Attributes
+  #     include ActiveModel::Enum
+  #     attribute :status, :integer
+  #     enum :status, active: 0, archived: 1
+  #   end
+  #
+  # Finally it's also possible to use a string column to persist the enumerated value.
+  #
+  #   class Conversation
+  #     include ActiveModel::Attributes
+  #     include ActiveModel::Enum
+  #     attribute :status, :integer
+  #     enum :status, active: "active", archived: "archived"
+  #   end
+  #
+  # Note that when an array is used, the implicit mapping from the values to database
+  # integers is derived from the order the values appear in the array. In the example,
+  # <tt>:active</tt> is mapped to +0+ as it's the first element, and <tt>:archived</tt>
+  # is mapped to +1+. In general, the +i+-th element is mapped to <tt>i-1</tt> in the
+  # database.
+  #
+  # Therefore, once a value is added to the enum array, its position in the array must
+  # be maintained, and new values should only be added to the end of the array. To
+  # remove unused values, the explicit hash syntax should be used.
+  #
+  # In rare circumstances you might need to access the mapping directly.
+  # The mappings are exposed through a class method with the pluralized attribute
+  # name, which return the mapping in a ActiveSupport::HashWithIndifferentAccess :
+  #
+  #   Conversation.statuses[:active]    # => 0
+  #   Conversation.statuses["archived"] # => 1
+  #
+  # Use that class method when you need to know the ordinal value of an enum.
+  # For example, you can use that when manually building SQL strings:
+  #
+  #   Conversation.where("status <> ?", Conversation.statuses[:archived])
+  #
+  # You can use the +:prefix+ or +:suffix+ options when you need to define
+  # multiple enums with same values. If the passed value is +true+, the methods
+  # are prefixed/suffixed with the name of the enum. It is also possible to
+  # supply a custom value:
+  #
+  #   class Conversation < ActiveRecord::Base
```

Comment:
Should we avoid having classes inherit from `ActiveRecord::Base` in the docs?

Reply:
Yes. I think we should mostly avoid these in ActiveModel.